### DISCUSSION TITLE
Simple Windows Switcher is not working

### DISCUSSION THREAD

**Discussion by piterq** [valinet/ExplorerPatcher#687] [General]
Hi.
I'm setting "Window switcher" option to "Simple Windows Switcher", but the effect is that Alt+Tab activates standard Windows NT windows switcher. I can see this errors in console when I'm setting this option in ExplorerPatcher properties (after setting this option, but before restarting file explorer):
```
====================================
An error occured in the application.
Error number: 0x80070581
Description: Klawisz dostŕpu jest ju┐ zarejestrowany.
Here is the stack trace:
[  0] = [0x00007FFF1CA1117A] ::
[  1] = [0x00007FFF1CA114E7] ::
[  2] = [0x00007FFF1CA16FC3] :: sws_WindowSwitcher_Initialize:
[  3] = [0x00007FFF1CA1FE67] :: UpdateHMDEmulationStatus:
[  4] = [0x00007FFF359B54E0] :: BaseThreadInitThunk:
[  5] = [0x00007FFF36CA485B] :: RtlUserThreadStart:
===================================
===================================
An error occured in the application.
Error number: 0x80070581
Description: Klawisz dostŕpu jest ju┐ zarejestrowany.
Here is the stack trace:
[  0] = [0x00007FFF1CA1117A] ::
[  1] = [0x00007FFF1CA114E7] ::
[  2] = [0x00007FFF1CA1FE6E] :: UpdateHMDEmulationStatus:
[  3] = [0x00007FFF359B54E0] :: BaseThreadInitThunk:
[  4] = [0x00007FFF36CA485B] :: RtlUserThreadStart:
====================================
```
Note: "Klawisz dostŕpu jest ju┐ zarejestrowany." means that access key (hotkey?) is already registered.
P.S. I've used switcheroo program in the past (it's windows switcher alternative software), so I launched it again and set in options "windows switching key" to something other than Alt+Tab (just to be sure that it isn't confilcting in any way). It didn't help.

**Comment by valinet** [comment]
Yeah, well, something registers that hotkey. Determine what that is, disable it, and it will work. The hotkey can be registered just fine on vanilla Windows installs, the time at which I register it in sws is before `explorer` registers it for its switcher, which is anyway superseded by sws. I haven't and don't test this against other third party software. The `RegisterHotKey` API fails like that if some program has already registered said hotkey, which is kind of expected. So yeah, make sure these are not registered in other programs: 

* `Alt`+`Tab`
* `Alt`+`Shift`+`Tab`
* `Alt`+`Ctrl`+`Tab`
* `Alt`+`Ctrl`+`Shift`+`Tab`
* `Alt`+`~`
* `Alt`+`Shift+~`
* `Alt`+`Ctrl`+`~`
* `Alt`+`Ctrl`+`Shift`+`~`

If you switch from Windows 10/11 switcher to sws, that error is expected: `explorer` already registered the hot key. The solution is to restart `explorer`, as the (*) near the setting label implies. After that, if the error still pops up, then it means something else still registers that key combination. And no, it's not the Windows NT switcher: that does not register any hot key, but it rather is implemented directly in the user mode portion of the keyboard driver, so to say, it hooks keys directly when they are filtered, providing a fallback when no hotkey is registered or when hotkeys are not working due to various reasons.
